paddle/framework/channel_test.cc

```diff
@@ -11,16 +11,58 @@ distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License. */
+#include <gtest/gtest.h>
 
 #include "paddle/framework/channel.h"
+#include "threadpool.h"
 
-#include "gtest/gtest.h"
+using paddle::framework::Channel;
+using paddle::framework::MakeChannel;
+using paddle::framework::CloseChannel;
 
 TEST(Channel, MakeAndClose) {
-  using paddle::framework::Channel;
-  using paddle::framework::MakeChannel;
-  using paddle::framework::CloseChannel;
+  Channel<int>* ch = MakeChannel<int>(10);
+  CloseChannel(ch);
+}
 
+TEST(Channel, Buffered) {
   Channel<int>* ch = MakeChannel<int>(10);
+
+  for (int i = 0; i < 10; ++i) {
+    ch->Send(&i);
+  }
+
+  int temp = -1;
+  for (int j = 0; j < 10; ++j) {
+    ch->Receive(&temp);
+    EXPECT_EQ(temp, j);
+  }
+  CloseChannel(ch);
+}
+
+TEST(Channel, MultiThread) {
+  namespace framework = paddle::framework;
+
+  const int capacity = 10;
+  Channel<int>* ch = MakeChannel<int>(capacity);
+
+  framework::ThreadPool* pool;
+  pool = framework::ThreadPool::GetInstance();
```

Comment:
For two reasons, let us do not use `framework::ThreadPool` in this unit test, but use `std::thread`:

1. To minimize the dependencies.
1. `framework::ThreadPool` sets the number of OS threads automatically, which could be one.